tracer/src/Datadog.Trace/RuntimeMetrics/RuntimeMetricsWriter.cs

```diff
@@ -158,6 +158,11 @@ public void Dispose()
             _exceptionCounts.Clear();
         }
 
+        internal void UpdateStatsd(IDogStatsd statsd)
+        {
+            _statsd = statsd;
```

Comment:
Given it's a background thread that's running here, I think we need an interlocked (or volatile read and write, but this is easier) to ensure cross-thread visibility?

```suggestion
            Interlocked.Exchange(ref _statsd, statsd);
```